### PR TITLE
Allow running on windows

### LINKED ISSUE #491: Support Windows
- [ ] Add windows tests in the CI
- [ ] Check that signals work the same way
- [ ] Check that envvars work the same way
- [ ] Adjust documentation
- [ ] Find a windows maintainer

### PR DESCRIPTION
**Pull request by spapas**
*Relates* to #491. This is a very simple PR that allows this projecto run on Windows. Let's keep it as draft until I am able to test it on my project. One thing to notice is that this completely ignores signals so we should probably document that somewhere.

### Successful PR Checklist:

- [ ] Tests
  - [x] Not needed
- [ ] Documentation
  - [ ] Where should we document this?

#### PR label(s): 
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20breaking%20%F0%9F%92%A5
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20feature%20%E2%AD%90%EF%B8%8F
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20bugfix%20%F0%9F%95%B5%EF%B8%8F
  - [x] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20miscellaneous%20%F0%9F%91%BE
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20dependencies%20%F0%9F%A4%96
  - [ ] https://github.com/procrastinate-org/procrastinate/labels/PR%20type%3A%20documentation%20%F0%9F%93%9A

### CODE REVIEW

**Comment by ewjoachim** [comment]
> Where should we document this?

I'd suggest a "how-to" page for "how to run procrastinate under Windows" ?

File: docs/howto/basics/windows.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Running on Windows
+
+Procrastinate has been tested and is known to work on Windows. However, there are
+a few things to keep in mind:
```
Comment (ewjoachim) [inline docs/howto/basics/windows.md:4]:
```suggestion
Procrastinate has been manually tested and is known to work on Windows. However,
there are a few things to keep in mind:
```

File: docs/howto/basics/windows.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+-  Due to the way signals are implemented on Windows (in short, they are not fully supported),
+  the worker will not be able to gracefully stop when receiving a `SIGINT` or `SIGTERM` signal
+  as described in the {doc}`../production/retry_stalled_jobs` guide. This means that any running
+  jobs will be halted abruptly.
+
+- We do not use windows for running Procrastinate in production, so we may not
+  be aware of all the issues that may arise.
+
```
Comment (ewjoachim) [inline docs/howto/basics/windows.md:13]:
```suggestion

- There are no automated tests in place for windows, so it's possible you encounter
  bugs. If you know what needs to be fixed, we accept
  [pull requests](https://github.com/procrastinate-org/procrastinate/pulls).
  Thank you!
```

**Review by ewjoachim — APPROVED** [review]
Just a couple of additions to the doc. Very nice job, thank you :)

**Comment by spapas** [comment]
Hello @ewjoachim thank you for your kind words! I'm gonna keep it as a draft for a little more to make some more tests on my project and then I'll convert it to normal PR. 

I don't have anything more for this PR but If you think that we can change/add anything please tell me.

**Comment by spapas** [comment]
Hello friends, I'm happy to report that I wasn't able to find any other problems with Windows and everything seems to be working great on my project so I'll change the to a normal PR.

**Comment by ewjoachim** [comment]
It's entirely my fault, but the doc is now missing a newline to separate the bullet list from the last paragraph, see here: https://procrastinate--1236.org.readthedocs.build/en/1236/howto/basics/windows.html

Can you fix this ?

**Comment by ewjoachim** [comment]
Perfect (yeah, sorry I forgot to follow-up too, otherwise I'd have done it)

Congratz with your PR !

**Pull request merged**